### PR TITLE
[Protocol3] Implement IUserStakingPool without a working auction

### PR DESCRIPTION
**Pull request by dong77**
This is a very basic implementation of user staking. Currently, all LRC are rewarded to users, which will be altered later.

### CODE REVIEW

**Comment by dong77** [comment]
@Brechtpd Please take a brief look. I'll add documentations later.

File: packages/loopring_v3/contracts/impl/StakingPool.sol [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+        )
+        external
+        onlyOwner
+        returns (
+            address auction
+        )
+    {
+        uint amountS = ERC20(tokenS).balanceOf(address(this));
+        require(amountS > 0, "ZERO_AMOUNT");
+
```
Comment (Brechtpd) [inline packages/loopring_v3/contracts/impl/StakingPool.sol]: Check tokenS != LRC

File: packages/loopring_v3/contracts/impl/StakingPool.sol [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        returns (uint minutes_)
+    {
+        if (users[user].depositedAt.add(MIN_WITHDRAW_DELAY) <= now) return 0;
+        else return users[user].depositedAt.add(MIN_WITHDRAW_DELAY).sub(now);
+    }
+
+    function userRewardWaitTime(address user)
+        view
+        private
+        returns (uint minutes_)
```
Comment (Brechtpd) [inline packages/loopring_v3/contracts/impl/StakingPool.sol]: Currently seconds

File: packages/loopring_v3/contracts/impl/StakingPool.sol [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+            auction
+        );
+    }
+
+    // -- Private Function --
+
+    function userWithdrawalWaitTime(address user)
+        view
+        private
+        returns (uint minutes_)
```
Comment (Brechtpd) [inline packages/loopring_v3/contracts/impl/StakingPool.sol]: Currently seconds

**Pull request merged**